### PR TITLE
Add push_image.yml, make some changes to plan/dev

### PR DESCRIPTION
**Pull request by dhansen-gw**
To build and deploy images in a generic way, hopefully.

I mashed this together from https://github.com/goodwatercap/company-tearsheet-web-client/blob/dev/.github/workflows/deploy.yml and https://github.com/goodwatercap/.github/blob/main/.github/workflows/apply.yml as part of the Redash deployment work.

---
While I was working on this, I also made a couple improvements regarding https://app.asana.com/0/1201289502507195/1202831989652774/f:
1. Used `tee` to print incremental output as the actions are running
2. Changed failure conditions: instead of `continue_on_error` (which moves the ❌ to the next step), let the step fail. Then the following steps run with `if: success() || failure()`, so that they still post comments when there are failures (see [StackOverflow](https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f) for why I chose this over `always()`)
3. Add a `merge_stage_prs` flag for the stage env, and inline the merge_prs script
4. Make `plan.yml` work more like `apply.yml`; it was doing some weird stuff with output files that was making failures harder to find.

### CODE REVIEW

File: .github/workflows/push_image.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+      # Get the GKE credentials, so we can apply against the cluster
+      - name: get-credentials
+        # Workaround to https://github.com/actions/runner/issues/520
+        env:
+          HAS_CREDENTIALS: ${{ secrets.k8s_cluster_id && secrets.k8s_cluster_zone }}
+        if: env.HAS_CREDENTIALS
+        run: |-
+          gcloud container clusters get-credentials ${{ secrets.k8s_cluster_id }} --zone ${{ secrets.k8s_cluster_zone }}
+
```
Comment (nikhname) [inline .github/workflows/push_image.yml:53]: Nit, extra space here

File: .github/workflows/push_image.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  workflow_call:
+    inputs:
+      dockerfile:
+        description: Name of the dockerfile to build and push
+        required: True
+        type: string
+    secrets:
+      repository:
+        description: Name of the repo within Container Registry to push into
+        required: True
```
Comment (nikhname) [inline .github/workflows/push_image.yml:13]: Nit - other true/false vars are lowercase but this not an issue for compiling

**Review by nikhname — APPROVED** [review]

File: .github/workflows/apply.yml [hunk 3]
```diff
@@ -117,16 +144,19 @@ jobs:
           cat <<- EOF >> terraform.tfvars.json
             ${{ inputs.tf_var_json }}
           EOF
-          output=$(terraform apply --auto-approve --input=false -no-color)
-          echo $output
+          terraform apply --auto-approve --input=false -no-color 2>&1 | tee tf_output.txt
```
Comment (nikhname) [inline .github/workflows/apply.yml:147]: Does tee create tf_output.txt if it doesn't exist?

**Review by nikhname — APPROVED** [review]

File: .github/workflows/apply.yml (hunk 3 above)
Comment (dhansen-gw) [reply .github/workflows/apply.yml:147]: Yes... I might need to clean up that file afterwards, though

**Pull request merged**